app/index.py

```diff
@@ -11,19 +10,50 @@
 app.secret_key = get_config().app_secret_key
 
 
+def authenticated(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if 'authenticated' not in session:
+            return redirect(url_for('login'))
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
 @app.route('/')
 @app.route('/index')
 def index():
     return render_template('base.html')
 
 
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    if request.method == 'POST':
+        if request.form['key'] == get_config().api_key:
+            session['authenticated'] = True
+            return redirect(url_for('view_all_data'))
+        else:
+            error_msg = 'Invalid password'
+        return render_template('login.html', error=error_msg)
+    return render_template('login.html')
+
+
+@app.route('/logout')
+@authenticated
+def logout():
+    session.pop('auth_key', None)
+    return redirect(url_for('index'))
```

Comment:
To powinno się znaleźć we własnej domenie na przykład login. Tak jak wygląda strucktua w domenie training_data, tylko tutaj tylko kontrolery potrzebujemy

Reply:
Chyba nie całkiem rozumiem, który dokładnie fragment. Endpointy do logowania przerzuciłem do `app.domain.common.authentication.routes.routes.py` - jakbyś mógł spojrzeć na to jeszcze raz i powiedzieć co powinno być gdzie indziej

Reply:
Teraz, jak przeniosłeś to z maina to jest git